src/main/java/jenkinsci/plugins/telegrambot/TelegramBotDelegate.java

```diff
@@ -29,14 +31,20 @@
     public void perform(Run<?, ?> run, FilePath filePath, Launcher launcher, TaskListener taskListener)
             throws IOException, InterruptedException {
 
-        String logMessage = run.getEnvironment(taskListener).expand(message);
+        String logMessage = message;
+        try {
+            logMessage = TokenMacro.expandAll(run, filePath, taskListener, message);
+        } catch (MacroEvaluationException e) {
+            LOGGER.log(Level.SEVERE, "Error while expanding the message", e);
+        }
 
         GlobalConfiguration config = GlobalConfiguration.getInstance();
 
         try {
+            String finalLogMessage = logMessage.substring(0, Math.min(logMessage.length(), 1024));
```

Comment:
I think we should add smth like "sorry, message has been cropped"

Reply:
to the message

Reply:
anyway let's keep commit atomicity and do it in another pr